src/Symfony/Component/Validator/Util/PropertyPath.php

```diff
@@ -42,7 +42,7 @@ public static function append($basePath, $subPath)
                 return $basePath.$subPath;
             }
 
-            return $basePath ? $basePath.'.'.$subPath : $subPath;
+            return '' !== (string) $basePath ? $basePath.'.'.$subPath : $subPath;
```

Comment:
I think the change is fine given the current implementation. If we want to make the method stricter, i.e. check the input more rigorously, that's the subject of another PR.